forte/pipeline.py

```diff
@@ -798,6 +810,11 @@ def add(
             # This will ask the job to keep a copy of the gold standard.
             self.evaluator_indices.append(len(self.components))
 
+        if ref_name is None:
+            self._ref_names.append("unknown")
+        else:
+            self._ref_names.append(ref_name)
```

Comment:
There is no check for duplications. What if we assign the same name to two processors? I think we should through an exception when they are being added.